### PR TITLE
Domains and Plans steps in Stepper

### PR DESCRIPTION
**Pull request by escapemanuele**
This PR recreates in Stepper the `Domains` and `Plans` steps from `/start`.

More about the motivation here: Project Thread pdDR7T-hL-p2
TLDR:  Signup flows are directly connected to our revenue, and they’re the first thing the user interacts with, yet it’s far too easy to break /start, Stepper is far more robust. Also, in the new tailored flows we need to jump back and forth between Stepper and /start(Domains and Plans).

The two new steps currently live in the Link in Bio flow, ready to be tested

## What's missing?

- [x] Understand the best way to manage if users go back after reaching checkout (so having already created a site, we want to avoid to create a site again). Currently user can go to processing step, that loops without a function to execute, or back to the `site-creation-step` that will bring the user back to checkout
- [x] Fix needs to be made for `decide later`
- [x] Processing step messages and loader must be as in start
- [x] Transfer/Connect should bring to Plans without `start with a free plan`

## Testing

- Checkout the branch
- `yarn start` from root folder
- Go to `http://calypso.localhost:3000/setup?flow=link-in-bio`
- Try the flow: free/paid domains, free/paid plans, `decide later`, filter domains, ...

### CODE REVIEW

File: packages/data-stores/src/onboard/types.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+export type DomainForm = {
+	lastQuery?: string;
+	subdomainSearchResults?: DomainSuggestion[] | null;
+	loadingResults?: boolean;
+	searchResults?: DomainSuggestion[] | null;
+	hideInitialQuery?: boolean;
+};
+
+export type CartItem = Partial< RequestCartProduct > & Pick< RequestCartProduct, 'product_slug' >;
```
Comment (sirbrillig) [inline packages/data-stores/src/onboard/types.ts]: If you already are importing `RequestCartProduct` here, could you just use [MinimalRequestCartProduct](https://github.com/Automattic/wp-calypso/blob/3e6c473d71c9877b5f3173c1b608bb9ffc96e368/packages/shopping-cart/src/types.ts#L219-L220) instead of recreating it as `CartItem`? It doesn't seem worth duplicating the type.

File: packages/onboarding/src/cart/index.tsx [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+		siteSlug,
+		isFreeThemePreselected,
+		themeSlugWithRepo,
+		flowName,
+		userIsLoggedIn,
+		productsList
+	);
+}
+
+const addPrivacyProtectionIfSupported = (
```
Comment (sirbrillig) [inline packages/onboarding/src/cart/index.tsx]: I may be wrong (I bow to the domain team @Automattic/nomado ) but I think that this function has no effect and could be removed. See https://github.com/Automattic/wp-calypso/pull/63292 where this behavior was removed from checkout and the reasons why.

File: packages/onboarding/src/cart/index.tsx [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+		themeSlugWithRepo,
+		flowName,
+		userIsLoggedIn,
+		productsList
+	);
+
+	return providedDependencies;
+};
+
+function prepareItemForAddingToCart( item: CartItem, lastKnownFlow?: string ) {
```
Comment (sirbrillig) [inline packages/onboarding/src/cart/index.tsx]: I think that this file could probably also use `MinimalRequestCartProduct` instead of `CartItem`.

File: client/landing/stepper/declarative-flow/link-in-bio.ts [hunk 4]
```diff
@@ -65,10 +85,35 @@ export const linkInBio: Flow = {
...
+						persistSignupDestination( destination );
+						setSignupCompleteSlug( providedDependencies?.siteSlug );
+						setSignupCompleteFlowName( flowName );
+						const returnUrl = encodeURIComponent(
+							`${ window.location.origin }/setup/launchpad?siteSlug=${ providedDependencies?.siteSlug }&flow=${ flowName }`
+						);
+
+						return window.location.assign(
+							`/checkout/${ encodeURIComponent(
+								providedDependencies?.siteSlug as string
```
Comment (sirbrillig) [inline client/landing/stepper/declarative-flow/link-in-bio.ts]: If it's possible that `providedDependencies.siteSlug` does not exist (which I assume is possible since this uses optional chaining?) then this URL would end up as `/checkout/undefined`, which is not great UX. Probably better is to provide an empty string fallback since `/checkout` is a valid route.

```suggestion
								providedDependencies?.siteSlug ?? ''
```

(This might be something to keep in mind everywhere you have `as string`.)

**Review by sirbrillig — COMMENTED** [review]
Bravo for working on this! I know it's a work-in-progress but I left some fairly minor comments that I hope are helpful.

File: packages/onboarding/src/cart/index.tsx (hunk 2 above)
Comment (escapemanuele) [reply packages/onboarding/src/cart/index.tsx]: Thanks for suggesting this, would be nice not to have to load all the `productsList`.
I see that this function adds `extra: {privacy: true}` to the cartItem when called, but I don't know if that is useful.
I hope @Automattic/nomado can give us some more insight :)

**Comment by escapemanuele** [comment]
Thanks a lot @sirbrillig, I addressed you comments and hope to can remove the `addPrivacyProtectionIfSupported` function 👍

File: client/signup/config/flows-pure.js [hunk 5]
```diff
@@ -129,8 +129,8 @@ export function generateFlows( {
 					dependencies.siteSlug
 				) }`,
 			description: 'Beginning of the flow to create a link in bio',
-			lastModified: '2022-08-16',
-			showRecaptcha: true,
+			lastModified: '2022-09-20',
+			Recaptcha: true,
```
Comment (heavyweight) [inline client/signup/config/flows-pure.js]: was this change intentional?

Comment (escapemanuele) [reply client/signup/config/flows-pure.js]: @renancarvalho I see you changed this 👍

Comment (renancarvalho) [reply client/signup/config/flows-pure.js]: Nope, will revert. thank you.

File: packages/onboarding/src/cart/index.tsx (hunk 2 above)
Comment (sirbrillig) [reply packages/onboarding/src/cart/index.tsx]: re: Privacy protection, I found a conversation I had with Nomado about it in Slack, which may be helpful: p1651677667181089-slack-C0BNMNMNG

Comment (escapemanuele) [reply packages/onboarding/src/cart/index.tsx]: That's so great to hear @sirbrillig, I'll remove the function then. Thanks a lot.

**Comment by niranjan-uma-shankar** [comment]
When the PR is ready, could you please ping MarTech for review? It would be good to get a few more eyes from the team. If there's opportunity to break the PR down to smaller, more focused PRs, that would be appreciated.

File: client/landing/stepper/declarative-flow/internals/steps-repository/domains/style.scss [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+		padding-top: 0;
+		padding-bottom: 16px;
+	}
+	.search-component.has-focus {
+		border-color: unset;
+		box-shadow: unset;
+	}
+
+	.domain-suggestion__action {
+		border-radius: 4px;
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/domains/style.scss:113]: This needs a
```suggestion
		border-radius: 4px;
		margin: 0;
```

File: client/landing/stepper/declarative-flow/internals/steps-repository/plans/index.tsx [hunk 7]
```diff
@@ -0,0 +1,29 @@
+import { StepContainer } from 'calypso/../packages/onboarding/src';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import PlansWrapper from './plans-wrapper';
+import type { Step } from '../../types';
+
+const plans: Step = function plans( { navigation } ) {
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/plans/index.tsx]: See why below:

```suggestion
const plans: Step = function plans( { navigation, flow } ) {
```

File: client/landing/stepper/declarative-flow/internals/steps-repository/plans/index.tsx [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+	const { submit, goBack } = navigation;
+
+	const handleSubmit = () => {
+		submit?.();
+	};
+
+	return (
+		<StepContainer
+			// For now it only supports LiB flows
+			stepName={ 'link-in-bio' }
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/plans/index.tsx]:
```suggestion
			stepName={ flow }
```

File: client/landing/stepper/declarative-flow/internals/steps-repository/plans/index.tsx [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+			goBack={ goBack }
+			isHorizontalLayout={ false }
+			isWideLayout={ true }
+			hideFormattedHeader={ true }
+			isLargeSkipLayout={ false }
+			hideBack={ true }
+			stepContent={ <PlansWrapper flowName={ 'link-in-bio' } onSubmit={ handleSubmit } /> }
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/plans/index.tsx]:
```suggestion
			stepContent={ <PlansWrapper flowName={ flow } onSubmit={ handleSubmit } /> }
```

File: client/landing/stepper/declarative-flow/internals/steps-repository/plans/plans-wrapper.tsx [hunk 10]
```diff
@@ -0,0 +1,237 @@
...
+	const locale = useLocale();
+	const { __ } = useI18n();
+
+	const isDesktop = useDesktopBreakpoint();
+	const stepName = 'plans';
+	const isReskinned = true;
+	const customerType = 'personal';
+	const isInVerticalScrollingPlansExperiment = true;
+	const planTypes = undefined;
+	const headerText = 'Choose a plan';
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/plans/plans-wrapper.tsx]: Should be translated.

File: client/landing/stepper/declarative-flow/internals/steps-repository/plans/plans-wrapper.tsx [hunk 11]
```diff
@@ -0,0 +1,237 @@
...
+			<QueryPlans />
+			<MarketingMessage path="signup/plans" />
+			<div className={ classes }>{ plansFeaturesSelection() }</div>
+		</div>
+	);
+};
+
+export default connect( ( state ) => {
+	return {
+		siteType: getSiteType( state ),
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/plans/plans-wrapper.tsx]: A bit confused by this, the site doesn't exist yet.

File: client/landing/stepper/declarative-flow/internals/steps-repository/plans/style.scss [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+
+	.gridicon {
+		margin: -2px 3px 0 0;
+		vertical-align: middle;
+	}
+}
+
+.plans-step__compare {
+	margin-top: 25px;
+}
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/plans/style.scss]: Is this doing anything?

File: client/landing/stepper/declarative-flow/internals/steps-repository/plans/style.scss [hunk 13]
```diff
@@ -0,0 +1,158 @@
...
+
+
+.signup__step.is-plans .formatted-header__subtitle .button.is-borderless {
+	padding: 0;
+	color: inherit;
+	font-size: inherit;
+	font-weight: inherit;
+	line-height: inherit;
+	text-decoration: underline;
+}
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/plans/style.scss]: Is this doing anything?

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-creation-step/styles.scss [hunk 14]
```diff
@@ -0,0 +1,24 @@
+@import "../style";
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/site-creation-step/styles.scss:1]: Kinda weird. Are we sure this isn't imported somewhere else?

File: client/landing/stepper/declarative-flow/link-in-bio.ts [hunk 15]
```diff
@@ -20,7 +26,16 @@ export const linkInBio: Flow = {
...
-		return [ 'intro', 'linkInBioSetup', 'patterns', 'processing', 'launchpad' ] as StepPath[];
+		return [
+			'intro',
+			'linkInBioSetup',
+			'domains',
+			'patterns',
+			'siteCreationStep',
+			'processing',
+			'launchpad',
+			'plans',
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/link-in-bio.ts]: It's not a must, but having the steps in order is better.
```suggestion
			'intro',
			'linkInBioSetup',
			'domains',
			'plans',
			'patterns',
			'siteCreationStep',
			'processing',
			'launchpad',
			
```

File: client/landing/stepper/declarative-flow/link-in-bio.ts [hunk 16]
```diff
@@ -29,7 +44,11 @@ export const linkInBio: Flow = {
 		const flowProgress = useFlowProgress( { stepName: _currentStep, flowName } );
 		setStepProgress( flowProgress );
 		const siteSlug = useSiteSlug();
-		const userIsLoggedIn = useSelect( ( select ) => select( USER_STORE ).isCurrentUserLoggedIn() );
+		const { userIsLoggedIn } = useSelect( ( select ) => {
+			return {
+				userIsLoggedIn: select( USER_STORE ).isCurrentUserLoggedIn(),
+			};
+		} );
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/link-in-bio.ts]: This should be reverted.

File: client/landing/stepper/declarative-flow/link-in-bio.ts [hunk 17]
```diff
@@ -53,9 +72,10 @@ export const linkInBio: Flow = {
 		function submit( providedDependencies: ProvidedDependencies = {} ) {
 			recordSubmitStep( providedDependencies, '', flowName, _currentStep );
 			const logInUrl = getStartUrl();
-
 			switch ( _currentStep ) {
 				case 'intro':
+					clearSignupDestinationCookie();
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/link-in-bio.ts:77]: Ideally, we should not need this cookie any more.

File: client/landing/stepper/index.tsx [hunk 18]
```diff
@@ -58,6 +59,7 @@ function initializeCalypsoUserStore( reduxStore: any, user: CurrentUser ) {
 	config.isEnabled( 'signup/inline-help' ) && reduxStore.dispatch( requestHappychatEligibility() );
 	reduxStore.dispatch( setCurrentUser( user ) );
 	reduxStore.dispatch( requestSites() );
+	reduxStore.dispatch( setCurrentFlowName( 'link-in-bio' ) ); //make it dynamic ofc.
```
Comment (alshakero) [inline client/landing/stepper/index.tsx]: Where do we use this value? We shouldn't rely on this store in Stepper at all.

File: client/landing/stepper/index.tsx [hunk 19]
```diff
@@ -136,6 +138,7 @@ window.AppBoot = async () => {
 	const reduxStore = createReduxStore( initialState, initialReducer );
 	setStore( reduxStore, getStateFromCache( userId ) );
 	setupLocale( user, reduxStore );
+	window.redux = reduxStore;
```
Comment (alshakero) [inline client/landing/stepper/index.tsx]: It would be great to get rid of this. Would it be hard?

File: client/my-sites/checkout/get-thank-you-page-url/index.ts [hunk 20]
```diff
@@ -134,7 +135,10 @@ export default function getThankYouPageUrl( {
 	// (and WP.com's) paid blocks Upgrade Nudge.
 	if ( redirectTo ) {
 		const { protocol, hostname, port, pathname, searchParams } = getUrlParts( redirectTo );
-
+		if ( isCalypsoLive() ) {
+			debug( 'is calypso live, so returning that', redirectTo );
+			return redirectTo;
+		}
```
Comment (alshakero) [inline client/my-sites/checkout/get-thank-you-page-url/index.ts]: I think we don't need this anymore after fixing the URL. cc @escapemanuele .

File: packages/data-stores/src/onboard/actions.ts [hunk 21]
```diff
@@ -314,6 +320,25 @@ export const setEditEmail = ( email: string ) => ( {
...
 } );
 
+export const setDomainForm = ( step: Record< string, string > ) => {
+	const lastUpdated = Date.now();
+
+	return {
+		type: 'SET_DOMAIN_FORM' as const,
+		step: { ...step, lastUpdated },
+	};
+};
```
Comment (alshakero) [inline packages/data-stores/src/onboard/actions.ts:398]: We should create an issue to remove every object state (vs primitive). We should be able to track every piece of info individually.

File: packages/onboarding/src/cart/index.tsx [hunk 22]
```diff
@@ -0,0 +1,267 @@
...
+	const siteId = siteCreationResponse?.blog_details.blogid;
+	const providedDependencies = {
+		siteId,
+		siteSlug,
+		domainItem,
+	};
+
+	if ( isTailoredSignupFlow( flowName ) ) {
+		await setupSiteAfterCreation( { siteId, flowName: flowName } );
+	}
```
Comment (alshakero) [inline packages/onboarding/src/cart/index.tsx]: Clean ✨

**Review by alshakero — CHANGES_REQUESTED** [review]
Left some comments.

File: client/my-sites/checkout/get-thank-you-page-url/index.ts (hunk 20 above)
Comment (sirbrillig) [reply client/my-sites/checkout/get-thank-you-page-url/index.ts]: 💯  

If you do need to modify this function, please also modify the tests to prove what you're trying to prevent and protect against future regressions (see the comments in this file). There may also be a better way to do it. All the code that creates redirect URLs already takes into account different host names and making an exception for calypso.live feels like it might be a hack to fix something more subtle.

File: packages/onboarding/src/cart/index.tsx [hunk 23]
```diff
@@ -0,0 +1,267 @@
+/* eslint-disable no-restricted-imports */
```
Comment (sirbrillig) [inline packages/onboarding/src/cart/index.tsx]: Is this a good idea? Calypso [has a dependency](https://github.com/Automattic/wp-calypso/blob/3511f6b53c787a8c5b8a2ed15f8ddbc575ce3d82/client/package.json#L49) on the `onboarding` package, and by doing this, the `onboarding` package will have a dependency on calypso, creating a circular dependency. I don't know a ton about bundling, but I'm pretty sure that's a bad thing.

It looks like you're adding this so that you can use the `cartManagerClient` from calypso, but you can recreate that using the `shopping-cart` package here if you like, or inject it as a dependency wherever this is used.

File: client/landing/stepper/declarative-flow/internals/steps-repository/domains/style.scss (hunk 6 above)
Comment (renancarvalho) [reply client/landing/stepper/declarative-flow/internals/steps-repository/domains/style.scss:113]: Thank you! next commit will have it.

File: client/landing/stepper/declarative-flow/internals/steps-repository/plans/index.tsx (hunk 8 above)
Comment (renancarvalho) [reply client/landing/stepper/declarative-flow/internals/steps-repository/plans/index.tsx]: I think here stepName will be always "plans". right ?

File: client/landing/stepper/declarative-flow/internals/steps-repository/plans/plans-wrapper.tsx (hunk 11 above)
Comment (renancarvalho) [reply client/landing/stepper/declarative-flow/internals/steps-repository/plans/plans-wrapper.tsx]: indeed, removed.

File: client/landing/stepper/declarative-flow/internals/steps-repository/plans/index.tsx (hunk 8 above)
Comment (alshakero) [reply client/landing/stepper/declarative-flow/internals/steps-repository/plans/index.tsx]: I vaguely remember some insanity where `stepName` meant flow name. Please check what this prop is used for, then decide how to handle it.

File: client/landing/stepper/declarative-flow/internals/steps-repository/plans/style.scss (hunk 13 above)
Comment (renancarvalho) [reply client/landing/stepper/declarative-flow/internals/steps-repository/plans/style.scss]: 👍  removed.

File: client/landing/stepper/index.tsx (hunk 19 above)
Comment (escapemanuele) [reply client/landing/stepper/index.tsx]: I used it for https://github.com/Automattic/wp-calypso/blob/update/tailored-flows-domains-step-backup/client/lib/signup/step-actions/fetch-sites-and-user.js#L11, but we're not currently using in the flow and it's working.
Maybe it was used to have a faster start for the user after the purchase? I think we can get rid of it here.

File: client/my-sites/checkout/get-thank-you-page-url/index.ts (hunk 20 above)
Comment (escapemanuele) [reply client/my-sites/checkout/get-thank-you-page-url/index.ts]: We do not need `isCalypsoLive` anymore luckily, removed 👍

File: packages/onboarding/src/cart/index.tsx (hunk 23 above)
Comment (escapemanuele) [reply packages/onboarding/src/cart/index.tsx]: Thank you @sirbrillig, I will take this into consideration and try to remove the circular dependency.

File: packages/data-stores/src/onboard/actions.ts (hunk 21 above)
Comment (escapemanuele) [reply packages/data-stores/src/onboard/actions.ts:398]: On it 👍

File: client/landing/stepper/index.tsx (hunk 19 above)
Comment (alshakero) [reply client/landing/stepper/index.tsx]: Yup nuke it.

**Comment by alshakero** [comment]
Hi @niranjan-uma-shankar! We started with two smaller PRs (plans and domains separately), but that proved even more complicated than this because of their entanglement with checkout. Both PRs had a considerable overlap in their changes.

We'll publish a CFT pretty soon, and we'll definitely ping your team.

File: client/landing/stepper/index.tsx (hunk 18 above)
Comment (renancarvalho) [reply client/landing/stepper/index.tsx]: I believe we don't need it at all, removing it.

File: client/landing/stepper/declarative-flow/internals/steps-repository/plans/index.tsx (hunk 7 above)
Comment (renancarvalho) [reply client/landing/stepper/declarative-flow/internals/steps-repository/plans/index.tsx]: Thank you!

File: client/landing/stepper/declarative-flow/internals/steps-repository/plans/style.scss (hunk 12 above)
Comment (renancarvalho) [reply client/landing/stepper/declarative-flow/internals/steps-repository/plans/style.scss]: nope, removed. It was from 'old code'
thank you

File: client/landing/stepper/declarative-flow/internals/steps-repository/plans/plans-wrapper.tsx [hunk 24]
```diff
@@ -0,0 +1,234 @@
...
+	const classes = classNames( 'plans-step', {
+		'in-vertically-scrolled-plans-experiment': isInVerticalScrollingPlansExperiment,
+		'has-no-sidebar': true,
+		'is-wide-layout': true,
+	} );
+
+	return (
+		<div className="stepper-plans">
+			<QueryPlans />
+			<MarketingMessage path="signup/plans" />
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/plans/plans-wrapper.tsx]: This seems fishy.

File: client/landing/stepper/declarative-flow/internals/steps-repository/processing-step/tailored-flow-precheckout-screen/style.scss [hunk 25]
```diff
@@ -0,0 +1,117 @@
...
+			height: 100%;
+			width: 100%;
+			position: fixed;
+			top: 0;
+			left: 0;
+			display: flex;
+			flex-direction: column;
+			justify-content: center;
+			align-items: center;
+		}
```
Comment (alshakero) [inline client/landing/stepper/declarative-flow/internals/steps-repository/processing-step/tailored-flow-precheckout-screen/style.scss:32]: This might not be needed here because it's a proper step here, not an overlay.

**Review by alshakero — COMMENTED** [review]
I did another read and audited the imports and things look better. I'm confused by the 124KB addition.

File: client/landing/stepper/declarative-flow/internals/steps-repository/plans/plans-wrapper.tsx (hunk 24 above)
Comment (escapemanuele) [reply client/landing/stepper/declarative-flow/internals/steps-repository/plans/plans-wrapper.tsx]: Removed `MarketingMessage`, as without a `siteId` it won't ever return any message.

**Review by heavyweight — APPROVED** [review]
Retested after a while and couldn't find any issues

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7635944

Hi @escapemanuele, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Help your Newsletter stand out with a custom domain. Not sure yet? <span>Decide later</span>.`
- `Set your Link in Bio apart with a custom domain. Not sure yet? <span>Decide later</span>.`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7635944).

File: client/landing/stepper/declarative-flow/internals/steps-repository/processing-step/index.tsx [hunk 26]
```diff
@@ -48,27 +53,29 @@ const ProcessingStep: Step = function ( props ) {
...
-					// We can now get the most up to date values from hooks inside the flow creating submit(),
-					// including the values that were updated during the action() running.
-					setDestinationState( destination );
-					setHasActionSuccessfullyRun( true );
-				} catch ( e ) {
-					// eslint-disable-next-line no-console
-					console.error( 'ProcessingStep failed:', e );
-					captureFlowException( e );
-					submit?.( {}, ProcessingResult.FAILURE );
+		if ( action ) {
```
Comment (arthur791004) [inline client/landing/stepper/declarative-flow/internals/steps-repository/processing-step/index.tsx:56]: Why do we need to check action here?

Comment (escapemanuele) [reply client/landing/stepper/declarative-flow/internals/steps-repository/processing-step/index.tsx:56]: I'm not sure if it still needed. It was a patch added before
`submit?.( {}, ProcessingResult.NO_ACTION );` was added, maybe. And it would have covered users hitting the back button from checkout. If we test it and it is not needed, then we can remove it 👍

Comment (escapemanuele) [reply client/landing/stepper/declarative-flow/internals/steps-repository/processing-step/index.tsx:56]: On a second thought, I think it was added because otherwise it would be difficult for users to go back in the flow after reaching checkout.

**Comment by niranjan-uma-shankar** [comment]
@escapemanuele FYI that this PR re-added the https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plan-comparison-with-highlights folder which was previously removed in https://github.com/Automattic/wp-calypso/pull/69710. It would be good to remove the folder since we don't use it anywhere.

**Comment by escapemanuele** [comment]
Thank you @niranjan-uma-shankar, I opened [ this PR ](https://github.com/Automattic/wp-calypso/pull/71487).

**Comment by ddc22** [comment]
I am transitioning over to the stepper framework from signup framework. I am still trying to figure out how we model `steps` in the stepper universe.

At the point in time this PR was merged a domain selection step which had the name [choose-a-domain](https://github.com/Automattic/wp-calypso/blob/b1ce8838ce99c13503b6be364e4c60daf9b519d6/client/landing/stepper/declarative-flow/internals/steps-repository/choose-a-domain/index.tsx) already existed, introduced by the following PR.

-  #67539

https://github.com/Automattic/wp-calypso/blob/b1ce8838ce99c13503b6be364e4c60daf9b519d6/client/landing/stepper/declarative-flow/internals/steps-repository/choose-a-domain/index.tsx#L43-L62

Should we not have re-used that step here @escapemanuele @alshakero ? The logic looks duplicated in most areas I checked. Personally I think creating new steps for each use case is going to be a maintenance nightmare. 

CC - @pgk @southp

**Comment by pgk** [comment]
> Should we not have re-used that step here @escapemanuele @alshakero ? The logic looks duplicated in most areas I checked. Personally I think creating new steps for each use case is going to be a maintenance nightmare.
> 
> CC - @pgk @southp

Agreed! How can we remove this duplication? Can we reuse the choose-a-domain step here?

**Comment by escapemanuele** [comment]
For sure @jdc91, removing duplication would be great. These 2 steps could be one 👍
